### PR TITLE
[JSON] Notifications on all JSON commands + del/strappend tests

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: src/notifications.c [hunk 1]
```diff
@@ -88,7 +97,17 @@ int HashNotificationCallback(RedisModuleCtx *ctx, int type, const char *event,
   else CHECK_CACHED_EVENT(rename_from)
   else CHECK_CACHED_EVENT(rename_to)
   else CHECK_CACHED_EVENT(loaded)
-//  else CHECK_CACHED_EVENT(json_set)
+  else CHECK_CACHED_EVENT(json_set)
```
Comment (rafie) [inline src/notifications.c:100]: Do we know this is kosher? I can spot "json_set" (for instance) in a couple of spots in RedisJSON and Rust might not unify the strings.

Comment (ashtul) [reply src/notifications.c:100]: You are right about this one.
@oshadmi, can we unify them on RedisJSON side?

Comment (oshadmi) [reply src/notifications.c:100]: @ashtul @rafie It seems that Rust is using the same address for similar string slices (`&str`), so both `"json_set"` literals are actually already being unified.
However, when calling the API `RedisModule_NotifyKeyspaceEvent`, we are constructing a new `CString` from the string slice (to work according to Rust FFI  (Foreign Function Interface)) and passing it to the API, so the notification callback could be receiving a different pointer each time for the `event` parameter and currently string comparison seems to be required.

Comment (ashtul) [reply src/notifications.c]: Done!

File: src/notifications.c [hunk 2]
```diff
@@ -222,6 +224,22 @@ int HashNotificationCallback(RedisModuleCtx *ctx, int type, const char *event,
       break;
   }
 
+  if (!strncmp(event, "json_", strlen("json_"))) {
+    if (!strncmp(event + JSON_LEN, "set", strlen("set")) ||
+        !strncmp(event + JSON_LEN, "del", strlen("del")) ||
+        !strncmp(event + JSON_LEN, "numincrby", strlen("numincrby")) ||
+        !strncmp(event + JSON_LEN, "nummultiby", strlen("nummultiby")) ||
```
Comment (oshadmi) [inline src/notifications.c:231]: Should compare with `multby`  (event name is `json_multby`)

And handle missing `json_numpowby` ?

File: src/notifications.c [hunk 3]
```diff
@@ -222,6 +224,22 @@ int HashNotificationCallback(RedisModuleCtx *ctx, int type, const char *event,
       break;
   }
 
+  if (!strncmp(event, "json_", strlen("json_"))) {
+    if (!strncmp(event + JSON_LEN, "set", strlen("set")) ||
+        !strncmp(event + JSON_LEN, "del", strlen("del")) ||
+        !strncmp(event + JSON_LEN, "numincrby", strlen("numincrby")) ||
```
Comment (oshadmi) [inline src/notifications.c:230]: Should compare with `incrby`  (event name is `json_incrby`)

**Review by oshadmi — CHANGES_REQUESTED** [review]

**Review by rafie — APPROVED** [review]

**Pull request merged**